pylint/checkers/dunder_methods.py

```diff
@@ -52,25 +52,33 @@ def open(self) -> None:
                 self._dunder_methods.update(dunder_methods)
 
     @staticmethod
-    def within_dunder_def(node: nodes.NodeNG) -> bool:
+    def within_dunder_or_lambda_def(node: nodes.NodeNG) -> bool:
         """Check if dunder method call is within a dunder method definition."""
         parent = node.parent
         while parent is not None:
             if (
                 isinstance(parent, nodes.FunctionDef)
                 and parent.name.startswith("__")
                 and parent.name.endswith("__")
+                or DunderCallChecker.is_lambda_rule_exception(parent, node)
             ):
                 return True
             parent = parent.parent
         return False
 
+    @staticmethod
+    def is_lambda_rule_exception(ancestor: nodes.NodeNG, node: nodes.NodeNG) -> bool:
+        return (
+            isinstance(ancestor, nodes.Lambda)
```

Comment:
We shouldn't backport this, since in 2.x of astroid, `FunctionDef` inherits from `Lambda`, see pylint-dev/astroid#2115.

Reply:
(And I don't think it's important enough to backport a different solution.)